synapse/handlers/message.py

```diff
@@ -1291,3 +1300,57 @@ def _expire_rooms_to_exclude_from_dummy_event_insertion(self):
                 room_id,
             )
             del self._rooms_to_exclude_from_dummy_event_insertion[room_id]
+
+    async def _rebuild_event_after_third_party_rules(
+        self, third_party_result: dict, original_event: EventBase
+    ) -> Tuple[EventBase, EventContext]:
+        # the third_party_event_rules want to replace the event.
+        # we do some basic checks, and then return the replacement event and context.
+
+        # Construct a new EventBuilder and validate it, which helps with the
+        # rest of these checks.
+        try:
+            builder = self.event_builder_factory.for_room_version(
+                original_event.room_version, third_party_result
+            )
+            self.validator.validate_builder(builder)
+        except SynapseError as e:
+            raise Exception(
```

Comment:
Each of these is a bare `Exception` (and not `SynapseError`) so that they'll bubble and show up in logs and return 500s?

Reply:
yes, exactly.